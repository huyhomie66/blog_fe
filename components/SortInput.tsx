import { Select, SelectProps } from "@chakra-ui/react";

interface SortOption {
  label: string;
  value: string;
}

const SORT_OPTIONS: SortOption[] = [
  { label: "Sort by Date (Newest first)", value: "date_desc" },
  { label: "Sort by Date (Oldest first)", value: "date_asc" }
];

interface SortInputProps extends Omit<SelectProps, "onChange"> {
  handleSortChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
  selectedValue: string;
}

function SortInput({ handleSortChange, selectedValue }: SortInputProps) {
  return (
    <Select width={"20vw"} value={selectedValue} onChange={handleSortChange}>
      {SORT_OPTIONS.map((option) => (
        <option key={option.value} value={option.value}>
          {option.label}
        </option>
      ))}
    </Select>
  );
}

export default SortInput;
