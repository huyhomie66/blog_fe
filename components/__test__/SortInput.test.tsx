// components/__test__/BlogBox.test.tsx
import { render } from "@testing-library/react";
import SortInput from "../SortInput";

describe("SortInput", () => {
  it("renders SortInput unchanged", () => {
    const { container } = render(
      <SortInput handleSortChange={() => {}} selectedValue="" />
    );
    expect(container).toMatchSnapshot();
  });
});
