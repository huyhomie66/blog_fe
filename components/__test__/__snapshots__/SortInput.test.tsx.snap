// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortInput renders SortInput unchanged 1`] = `
<div>
  <div
    class="chakra-select__wrapper css-g5kil8"
  >
    <select
      class="chakra-select css-1ik61og"
    >
      <option
        value="date_desc"
      >
        Sort by Date (Newest first)
      </option>
      <option
        value="date_asc"
      >
        Sort by Date (Oldest first)
      </option>
    </select>
    <div
      class="chakra-select__icon-wrapper css-162mkon"
    >
      <svg
        aria-hidden="true"
        class="chakra-select__icon"
        focusable="false"
        role="presentation"
        style="width: 1em; height: 1em; color: currentColor;"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
